Misc/NEWS.d/next/Library/2024-07-21-18-03-30.gh-issue-122088.vi2bP-.rst

```diff
@@ -0,0 +1,3 @@
+``@warnings.deprecated`` now copies to coroutine flag of functions and
+methods so that ``inspect.iscoroutinefunction()`` returns the correct
+result.
```

Comment:
```suggestion
:func:`@warnings.deprecated <warnings.deprecated>` now copies the
:data:`coroutine flag <inspect.CO_COROUTINE>` flag of functions and
methods so that :func:`inspect.iscoroutinefunction` returns the correct
result.
```

I don't where the 80-char limit is :D By the way, I think we need a "deco" role for adding the '@' automatically.

Reply:
cc @AA-Turner for the 'deco' role (I'm going offline now)